### DISCUSSION TITLE
Feature Request: juliapkg - Tables.jl integration

### DISCUSSION THREAD

**Discussion by chris-b1** [duckdb/duckdb#6407] [Ideas]
Rather than depending on / requiring a DataFrame, it may be better to only depend on Tables.jl and make table registration accept any compatible source.  Skimming the integration code here, it seems like it would be achievable to genericize this code a bit without any loss of performance.

https://github.com/duckdb/duckdb/blob/6b7362d2ca06e3085fb939ca958d1d169f10b6e0/tools/juliapkg/src/data_frame_scan.jl

some additional backround in this discourse thread
https://discourse.julialang.org/t/write-a-table-into-sqlite-and-duckdb-in-efficient-manner/93417/13

**Comment by Mytherin** [comment]
That sounds like a good idea. Happy to review a PR!

**Comment by aplavin** [comment]
https://github.com/duckdb/duckdb/pull/7984
